### PR TITLE
Move GPG signing to after installer generation so installers are signed too

### PR DESCRIPTION
**Pull request by sxa**
Self-explanatory. The existing solution does not GPG sign the installers as they haven't been created at the time the GPG signing is executed. This moves it to a more suitable location, and also separtes the operation into its own function.

Signed-off-by: Stewart X Addison <sxa@redhat.com>

### CODE REVIEW

**Comment by sxa** [comment]
run tests

**Review by andrew-m-leonard — APPROVED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1747,6 +1814,12 @@ class Build {
                         throw new Exception("[ERROR] Installer job timeout (${buildTimeouts.INSTALLER_JOBS_TIMEOUT} HOURS) has been reached OR the downstream installer job failed. Exiting...")
                     }
                 }
+                try {
+                    gpgSign()
+                    }
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1819]: one too many `}`?

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by sxa** [comment]
run tests

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 1 above)
Comment (sxa) [reply pipelines/build/common/openjdk_build_pipeline.groovy]: Yup ... Fixed. The linter inevitably picked that up too.

**Review by karianna — APPROVED** [review]

**Comment by sxa** [comment]
Merging this despite not all of the build pipeline tests having been completed on the basis that the Linux and Mac tests have completed and succesfully executed the sign job as well as they did before, and I want to be able to test this in a real pipeline with the installers enabled to verify that the fix does what it's supposed to

**Comment by sxa** [comment]
Testing at https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/

**Pull request merged**